### PR TITLE
Update Filter SearchField styling

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

| [Before](https://storybook.polaris.shopify.com/?path=/story/all-components-indexfilters--default&globals=polarisSummerEditions2023:true) | [After](https://5d559397bae39100201eedc1-scqfltzzwk.chromatic.com/?path=/story/all-components-indexfilters--default&globals=polarisSummerEditions2023:true) |
|--------|--------|
| Screenshot 2023-07-31 at 12 21 00 PM|Screenshot 2023-07-31 at 12 24 36 PM |

### How to 🎩

[Spin](https://admin.web.filter-search-2.kyle-durand.us.spin.dev/store/shop1/orders?inContextTimeframe=none)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/stupid-grapes-leave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Rebuilt Filter SearchField
```
Comment (chloerice) [inline .changeset/stupid-grapes-leave.md]:
```suggestion
Rebuilt `Filters` `SearchField`
```

**Review by chloerice — APPROVED** [review]
Looks much better! Not sure why the node install failed in the regression test build, maybe a rebase will help if the rerun doesn't? 🤔

**Review by laurkim — APPROVED** [review]
💯 

Small thing I noticed is that when it's disabled, it's using the default background color on the input field so it's not as obvious it's disabled next to other disabled elements (i.e., buttons). Might be worth adding disabled styles to the input so that it uses the `--p-color-bg-transparent-disabled-experimental` token to match the color on the buttons but that can also be a follow up PR

**Pull request merged**